sonar-db/src/test/java/org/sonar/db/permission/PermissionTemplateDaoTest.java

```diff
@@ -227,17 +222,57 @@ public void should_remove_group_permission_with_null_name() {
   }
 
   @Test
-  public void should_retrieve_permission_template() {
-    PermissionTemplateDto permissionTemplateDto = new PermissionTemplateDto().setName("Test template").setUuid("test_template");
-    PermissionTemplateDto templateWithPermissions = new PermissionTemplateDto().setUuid("test_template");
-    underTest = mock(PermissionTemplateDao.class);
-    when(underTest.selectByUuid(db.getSession(), "test_template")).thenReturn(permissionTemplateDto);
-    when(underTest.selectByUuidWithUserAndGroupPermissions(db.getSession(), "test_template")).thenReturn(templateWithPermissions);
-    when(underTest.selectPermissionTemplateWithPermissions(db.getSession(), "test_template")).thenCallRealMethod();
+  public void new_permission_template_with_empty_user_group_characteristics() {
+    PermissionTemplateDto template = underTest.insert(dbSession, newPermissionTemplateDto().setUuid("TEMPLATE_UUID"));
+
+    PermissionTemplateDto result = underTest.selectByUuidWithUserAndGroupPermissions(dbSession, "TEMPLATE_UUID");
+
+    assertThat(result).extracting(PermissionTemplateDto::getId, PermissionTemplateDto::getUuid, PermissionTemplateDto::getName, PermissionTemplateDto::getDescription)
+      .containsExactly(template.getId(), template.getUuid(), template.getName(), template.getDescription());
+
+    assertThat(result.getUsersPermissions()).isEmpty();
+    assertThat(result.getGroupsPermissions()).isEmpty();
+    assertThat(result.getCharacteristics()).isEmpty();
+  }
+
+  @Test
+  public void unknown_permission_template() {
+    PermissionTemplateDto result = underTest.selectByUuidWithUserAndGroupPermissions(dbSession, "UNKNOWN_TEMPLATE_UUID");
 
-    PermissionTemplateDto permissionTemplate = underTest.selectPermissionTemplateWithPermissions(db.getSession(), "test_template");
+    assertThat(result).isNull();
+  }
 
-    assertThat(permissionTemplate).isSameAs(templateWithPermissions);
+  @Test
+  public void permission_template_with_user_group_and_characteristics() {
```

Comment:
this test is not easily readable, it would be great to some some helpers methods.